src/components/basic/Badge.tsx

```diff
@@ -26,14 +27,23 @@ const Comp = styled.span<{ isRead: boolean; isNumber: boolean }>`
 `;
 
 const Badge = forwardRef<HTMLElement, BadgeProps>(function BadgeFn({ type, value, ...rest }, ref) {
+	const MAX_VALUE = 999;
 	const isNumber = useMemo(() => typeof value === 'number', [value]);
-	const badgeText = useMemo(() => (isNumber && value > 99 ? '99+' : value), [value, isNumber]);
+	const badgeText = useMemo(
+		() => (isNumber && value > MAX_VALUE ? `${MAX_VALUE}+` : value),
+		[value, isNumber]
+	);
 	const isRead = useMemo(() => type === 'read', [type]);
+	const showTooltip = useMemo(() => isNumber && value > MAX_VALUE, [value, isNumber]);
 
 	return (
-		<Comp ref={ref} isRead={isRead} isNumber={isNumber} {...rest}>
-			{badgeText}
-		</Comp>
+		<>
```

Comment:
Now that we have a single element returned, this fragment is not required anymore